src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -317,6 +369,62 @@ else if (WIFSIGNALED(status)) {
       return this;
    }
 
+   private int spawnWithCwdSetViaPthreadChdir(
+      IntByReference restrict_pid,
+      String restrict_path,
+      Pointer file_actions,
+      Pointer /*const posix_spawnattr_t*/restrict_attrp,
+      StringArray /*String[]*/argv,
+      Pointer /*String[]*/envp,
+      Path cwd)
+   {
+     int cwdBufSize = 1024;
+     long peer = Native.malloc(cwdBufSize);
+     Pointer oldCwd = new Pointer(peer);
+     LibC.getcwd(oldCwd, cwdBufSize);
+     String newCwd = cwd.toAbsolutePath().toString();
+     int rc = LibC.SYSCALL.syscall(LibC.SYSCALL.SYS___pthread_chdir, newCwd);
+     checkReturnCode(rc, "syscall(SYS__pthread_chdir) failed to set current directory");
+
+     try {
+       return LibC.posix_spawnp(restrict_pid, restrict_path, file_actions, restrict_attrp, argv, envp);
+     } finally {
+       rc = LibC.SYSCALL.syscall(LibC.SYSCALL.SYS___pthread_chdir, oldCwd);
+       Native.free(Pointer.nativeValue(oldCwd));
+       checkReturnCode(rc, "syscall(SYS__pthread_chdir) failed to restore current directory");
+     }
+   }
+
+   private int spawnWithCwdSetViaLinuxCwdChangeableExecutorService(
```

Comment:
The code is self-explanatory, so the method does not need to be so descriptive.  ``spawnLinuxWithCwd()``.